File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        sum=nums[0]
        ans=nums[0]
        for i in range(1,len(nums)):
            if nums[i]>nums[i-1]:
                sum+=nums[i]
            else:
                ans=max(sum,ans)
                sum=nums[i]
        ans=max(ans,sum)
        return ans
        

File: 1927-maximum-ascending-subarray-sum/1927-maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int ans=nums[0];
        int sum=nums[0];
        if(nums.size()==1) return nums[0];
        for(int i=1;i<nums.size();i++){
            if(nums[i]>nums[i-1]){
                sum+=nums[i];
            }
            else{
                ans=max(sum,ans);
                sum=nums[i];
            }

        }
        ans=max(sum,ans);
        return ans;
        
    }
};